src/test/java/org/folio/rest/api/InstanceBulkTest.java

```diff
@@ -0,0 +1,458 @@
+      ExecutionException,
+      TimeoutException {
+
+    int totalMoons = 20;
+    int expectedMatches = 11;
+    Map<String, JsonObject> moons = manyMoons(totalMoons,
+      InstanceBulkIdsGetField.ID,
+      InstanceBulkIdsGetFormat.RAW);
+    createManyMoons(moons);
+
+    String query = "query=keyword%20all%20%22Moon%20%231%2A%22";
+    URL getInstanceUrl = instanceBulkUrl("/ids?type=id&format=raw&" + query);
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+    client.get(getInstanceUrl, TENANT_ID, json(getCompleted));
+
+    Response response = getCompleted.get(5, SECONDS);
+    validateMoonsResponse(response, expectedMatches, moons);
+  }
+
+  @Test
+  public void canGetInstanceBulkOfRawIdWithLanguageGer()
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+
+    int totalMoons = 10;
+    int expectedMatches = 7;
+    Map<String, JsonObject> moons = manyMoons(totalMoons,
+      InstanceBulkIdsGetField.ID,
+      InstanceBulkIdsGetFormat.RAW);
+    createManyMoons(moons);
+
+    String query = "query=languages%3D%22ger%22";
+    URL getInstanceUrl = instanceBulkUrl("/ids?type=id&format=raw&" + query);
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+    client.get(getInstanceUrl, TENANT_ID, json(getCompleted));
+
+    Response response = getCompleted.get(5, SECONDS);
+    validateMoonsResponse(response, expectedMatches, moons);
+  }
+
+  private void validateMoonsResponse(Response response, int expectedMatches,
+      Map<String, JsonObject> moons) {
+    assertThat(response.getStatusCode(), is(HTTP_OK));
+    validateMoonsResponse(response.getJson(), expectedMatches, moons);
+  }
+
+  private void validateMoonsResponse(JsonObject collection,
+      int expectedMatches, Map<String, JsonObject> moons) {
+    JsonArray ids = collection.getJsonArray("ids");
+    assertThat(ids.size(), is(expectedMatches));
+    assertThat(collection.getInteger("totalRecords"), is (expectedMatches));
+
+    for (int i = 0; i < ids.size(); i++) {
+      JsonObject idObject = ids.getJsonObject(i);
+      assertThat(moons.containsKey(idObject.getString("id")), is(true));
+    }
+  }
+
+  private void setInstanceSequence(long sequenceNumber) {
+    final Vertx vertx = StorageTestSuite.getVertx();
+    final PostgresClient postgresClient =
+        PostgresClient.getInstance(vertx, TENANT_ID);
+    final CompletableFuture<Void> sequenceSet = new CompletableFuture<>();
+
+    vertx.runOnContext(v -> {
+      postgresClient.selectSingle("select setval('hrid_instances_seq',"
+          + sequenceNumber + ",FALSE)", r -> {
+            if (r.succeeded()) {
+              sequenceSet.complete(null);
+            } else {
+              sequenceSet.completeExceptionally(r.cause());
+            }
+          });
+    });
+
+    try {
+      sequenceSet.get(2, SECONDS);
+    } catch (Exception e) {
+      fail(e.getMessage());
```

Comment:
This does not swallow the stack trace:
```suggestion
      fail(e);
```